test/job.js

```diff
@@ -10,66 +10,45 @@ const delay = require("delay");
 const sinon = require("sinon");
 const { Job } = require("../dist/job");
 const { Agenda } = require("../dist");
+const getMongoCfg = require("./fixtures/mongo-connector");
 
-const mongoHost = process.env.MONGODB_HOST || "localhost";
-const mongoPort = process.env.MONGODB_PORT || "27017";
-const agendaDatabase = "agenda-test";
-const mongoCfg =
-  "mongodb://" + mongoHost + ":" + mongoPort + "/" + agendaDatabase;
+let mongoCfg;
 
 // Create agenda instances
 let agenda = null;
-let mongoDb = null;
 let mongoClient = null;
 
-const clearJobs = () => {
-  return mongoDb.collection("agendaJobs").deleteMany({});
-};
-
-// Slow timeouts for CI
-const jobTimeout = 500;
+// Slow timeouts for Travis
+const jobTimeout = process.env.TRAVIS ? 2500 : 500;
 const jobType = "do work";
 const jobProcessor = () => {};
 
 describe("Job", () => {
-  beforeEach((done) => {
-    agenda = new Agenda(
-      {
-        db: {
-          address: mongoCfg,
-        },
+  beforeEach(async () => {
+    mongoCfg = await getMongoCfg();
+  });
+
+  beforeEach(async () => {
+    agenda = new Agenda({
+      db: {
+        address: mongoCfg,
       },
-      async (error) => {
-        if (error) {
-          done(error);
-        }
+    });
 
-        try {
-          const client = await MongoClient.connect(mongoCfg, {
-            useNewUrlParser: true,
-            useUnifiedTopology: true,
-          });
```

Comment:
I'm not clear on the order of changes WRT the mongo connect options. Were these part of new tests/functionality or are they safe to ignore?

Reply:
They are useful for old mongodb versions. I think prior to 3.6. But we do not support older mongodb any more. So, it's okay to drop these options.